### ISSUE TITLE
Question about the UBody train/test sets

### ISSUE THREAD

**Issue by fabienbaradel** [IDEA-Research/OSX#61]
Hi,
Thanks for release the UBody dataset, that's very useful for the research community.
I am preprocessing your dataset into the train and test split. Below are some stats that I obtained.

| Split | Version | Number of frames | Total number of persons | 
| -- | -- | -- | -- |
| ```train``` | ```intra``` | 542'457 | 713'347 |
| ```test``` | ```intra``` | 200'581 | 235'060 |
| ```train``` | ```inter``` | 480'148 | 610'019 |
| ```test``` | ```inter``` | 262'890 | 338'388|

I am following your preprocessing steps [here](https://github.com/IDEA-Research/OSX/blob/13df1cbf9cc5ae72deb46ce7ab5832e7bcb8bd9d/data/UBody/UBody.py#L154) and as soon as ```valid_label > 0``` I keep the frame.
The good point is that the training sets are quite big, so that's very useful.
However the test set are super big (~200k frames!) so computing metrics on these test sets is taking a lot of time.
Am I doing something wrong? Should I keep only ```valid_label=2``` for the test sets?
Thanks for your help,

**Comment by fabienbaradel** [comment]
Hi,
Re-opening the issue to have a clarification. Some fits on the test set are not very good (which is absolutely normal - cf screenshot).
So when there are multiple person on the image at test time do you compute the metrics for all persons or only for the closest person to the camera?
Thanks for your help,

![Screenshot 2023-09-25 at 13 54 01](https://github.com/IDEA-Research/OSX/assets/11565723/fd333043-b4d1-475b-a065-a17f88eb9769)

**Comment by linjing7** [comment]
Hi, we compute the metrics for all the valid persons. During our manual checking, we annotate a valid label for each person. Thus, when there are multiple person on the image, we compute the metrics for the valid persons only.

**Comment by linjing7** [comment]
Hi, if `valid_label>0`, I would compute the vertices and joints errors on those located within the image, i.e., projecting the GT vertices into the image plane, and then if a certain vertice locates on the image, I would calculate its error value. 
https://github.com/IDEA-Research/OSX/blob/13df1cbf9cc5ae72deb46ce7ab5832e7bcb8bd9d/data/UBody/UBody.py#L602

**Comment by fabienbaradel** [comment]
Thanks!
